click/testing.py

```diff
@@ -278,6 +279,14 @@ def invoke(self, cli, args=None, input=None, env=None,
                 if e.code != 0:
                     exception = e
                 exit_code = e.code
+                # Fix for issue #362: `SystemExit.code` sometimes contains
+                # the error message, not the exit code.
+                try:
+                    if exit_code != int(exit_code):
+                        raise ValueError
+                except ValueError:
```

Comment:
Floating-point exit codes don't make sense, also this will trigger probably unexpected behavior with strings like "1" or even "0xfff".

We should mimick the behavior `sys.exit` exposes.

On 20 June 2015 11:00:29 CEST, Torf <notifications@github.com> wrote:
>> @@ -278,6 +279,14 @@ def invoke(self, cli, args=None, input=None,
>env=None,
>>                  if e.code != 0:
>>                      exception = e
>>                  exit_code = e.code
>> +                # Fix for issue #362: `SystemExit.code` sometimes
>contains
>> +                # the error message, not the exit code.
>> +                try:
>> +                    if exit_code != int(exit_code):
>> +                        raise ValueError
>> +                except ValueError:
>
>I'd rather not tie this to a specific data type (and if so,
>`numbers.Integral` would probably be a better choice). The current
>implementation works for any kind of number, including integers stored
>in a `float`.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/mitsuhiko/click/pull/363/files#r32883461